core/src/main/java/com/graphhopper/routing/profiles/TagParserFactory.java

```diff
@@ -0,0 +1,141 @@
+package com.graphhopper.routing.profiles;
+
+import com.graphhopper.reader.ReaderWay;
+import com.graphhopper.routing.util.AbstractFlagEncoder;
+import com.graphhopper.util.EdgeIteratorState;
+
+public class TagParserFactory {
```

Comment:
What do you mean with 'put this into classes'? Creating a RoundaboutTagParser and a HighwayTagParser etc? (This would introduce a lot more code/overhead IMO :)) 

I'm not super happy with it now but for me this factory was just convenient to avoid creating the same parsers in tests again and again.